test/parallel/test-diagnostics-channel-pub-sub.js

```diff
@@ -42,3 +42,10 @@ assert.ok(!dc.unsubscribe(name, subscriber));
 assert.throws(() => {
   dc.subscribe(name, null);
 }, { code: 'ERR_INVALID_ARG_TYPE' });
+
+// Reaching zero subscribers should not delete from the channels map as there
+// will be no more weakref to incRef if another subscribe happens while the
+// channel object itself exists.
+channel.subscribe(subscriber);
```

Comment:
Are there ever any potential issues with this stuff happening across multiple stacks?

```js
channel.subscribe(subscriber);
setImmediate(() => { channel.unsubscribe(subscriber); });
setImmediate(() => { channel.subscribe(subscriber); });
```

Reply:
Might have been before, but not after this change. In any case, the sync version reproduced the bug more easily.